### PR TITLE
Tell swapon to reinitialize swap if needed

### PR DESCRIPTION
**Pull request by dtardon**
If the page size of a swap space doesn't match the page size of the currently running kernel, `swapon` will fail. Let's instruct it to reinitialize the swap space instead.

This option exists since util-linux 2.15, released in 2009, hence it should be safe to use unconditionally.

### CODE REVIEW

**Comment by poettering** [comment]
Under the assumption this is just about the page size looks OK. (i.e. "-f" sounds a lot like it could destroy relevant stuff)

I hope this doesn't change the uuid of the swap?

**Comment by poettering** [comment]
@karelzak can you comment on this? is this safe to just do?

File: src/core/swap.c
Comment (keszybz) [inline src/core/swap.c]: Please write this as `--fixpgsz` for avoidance of doubt.

**Comment by keszybz** [comment]
This looks pretty safe according to the man page. But let's wait for confirmation from @karelzak before merging.

File: src/core/swap.c
Comment (dtardon) [reply src/core/swap.c]: That was actually my first choice. I only used short option because systemd seems to prefer that (judging from other calls of `exec_command_set` and `exec_command_append`).

**Review by yuwata — APPROVED** [review]

**Comment by yuwata** [comment]
> I hope this doesn't change the uuid of the swap?

It looks the command keeps uuid and label.
https://github.com/util-linux/util-linux/blob/db5aa5e5d8932c73f1b9f01fe567fa343898b825/sys-utils/swapon.c#L318-L351

**Comment by keszybz** [comment]
`--fixpgsz` was added in 0d761944dd1130c81ec770ed318ae7ed3338c702, 2009, we can assume it's available everywhere. Our tests don't check cross-arch swap shenanigans, so I don't think it makes sense wait more. Ubuntu CI seems to be having hardware issues.

**Comment by bluca** [comment]
> `--fixpgsz` was added in 0d761944dd1130c81ec770ed318ae7ed3338c702, 2009, we can assume it's available everywhere. Our tests don't check cross-arch swap shenanigans, so I don't think it makes sense wait more. Ubuntu CI seems to be having hardware issues.

Well, turns out not everywhere. Busybox does not support this option, so given this is hard-coded it makes it impossible to use swap when busybox is in use on a system. @dtardon this looks like a regression to me, can we revert this in 253 and then maybe rework it to be optional in 254?

**Comment by poettering** [comment]
Nah, we do not support busybox. Readme says so clearly. We require util-linux/libmount/mount/swapon which are close siblings maintained by the same people in the same codebase, sharing semantics and being closely intertwined. Let's not revert stuff for cases that are clearly against our documented requirements and are frankendistros that mix and match even closely related binaries. Swapon/mount are so closely related that they share the same config file (/etc/fstab) and library to read it (libmount).

**Comment by karelzak** [comment]
> This looks pretty safe according to the man page. But let's wait for confirmation from @karelzak before merging.

Sorry, for the delay :-)  Yes, it's safe. It calls mkswap with -U and -L to keep the label and UUID.

**Pull request merged**
